doc/sf/jacobi_theta.qbk

```diff
@@ -32,7 +32,7 @@ more accurate.
 
 Most applications will want to use the /q/ parameterization of the functions: `__jacobi_theta1`, `__jacobi_theta2`, `__jacobi_theta3`, and `__jacobi_theta4`, where /q/ is restricted to the domain (0, 1). However, a second [tau] parameterization is provided for all four functions, where
```

Comment:
Is there equivalent Mathematica syntax that has the same nome/tau/scaling conventions? It might be nice to put that in here  . . .

Reply:
I don't think Mathematica provides a tau parameterization. I'll add a link in the docs to EllipticTheta, though.

Reply:
I'm getting more confused about the `tau` parametrization. It now looks to me like you are trading a multiplication on each iteration for a logarithm and a call to `std::exp` on each iteration.  . .

Reply:
Implementing purely in `q`, without a `log`, would require a `pow` in place of the `exp`. The private `IMAGINARY` code paths have to do the `exp` anyway since they are implementing hyperbolic sine and cosine.

Exposing `tau` has the added benefit of letting the user provide a pre-logged argument for the sake of precision, e.g. with `q` close to 1 (so the user can use `log1p`) or if the argument provided would have been an exponential such as `exp(-a)`.

Reply:
You don't need a call to `pow`, since q^(n+1)^2 = q^n^2*qn*qn*q. Here's some pseudocode:

```cpp
template<typename Real>
Real new_theta1(Real x, Real q)
{
    using std::sqrt;
    using std::sin;
    using std::abs;
    Real scale = 2*sqrt(sqrt(q));
    Real term = sin(x);
    Real sum = term;
    size_t n = 1;
    Real qn = q;
    Real qnsq = q;
    Real prefactor = qnsq*qn;
    while (prefactor > abs(sum)*std::numeric_limits<Real>::epsilon())
    {
        term = prefactor*sin((2*n+1)*x);
        if (n & 1) {
            sum -= term;
        } else {
            sum += term;
        }
        qn *= q;
        qnsq = qnsq*q*qn*qn;
        prefactor = qnsq*qn;
        ++n;
    }
    return scale*sum;
}
```

Reply:
Very cool! Happy to explore performance improvements after everything else is settled.